London | Elhadj Abdoul Diallo | Module-User-Focused-Data | Week 1

eediallo opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

Create a web page that contains a header and three articles. Each article comprises:
- An image
- A title
- A summary and
- a link
The first article is styled using the:first-child pseudo-class which makes it stand out from the others.

## Questions
Dear Reviewer, thank you for dedicating your time to review my code!
I wrapped the articles inside a div tag and used the article tag to wrap the content (image, title, summary, link) of each article. 
Is that something acceptable in this coursework given that they ask to use semantic HTML?  I understand that we should not use div or span to write content straightaway.
But I am not sure about using them to wrap elements.

eediallo committed:
adds skeleton page

eediallo committed:
styles skeleton page

eediallo committed:
adds title of webpage

eediallo committed:
adds description of webpage

eediallo committed:
bolding 'Essentials' , adds article one image and footer

eediallo committed:
change hero-section width

eediallo committed:
uploads first article image

eediallo committed:
set img width to 100%

eediallo committed:
adds first article title

eediallo committed:
adds first article summary

eediallo committed:
replace div tag with header tag and add link to first article

eediallo committed:
adds content of first article

eediallo committed:
changes selector

eediallo committed:
changes layout of webpage

eediallo committed:
updates layout style

eediallo committed:
adds first: child pseudo class to style the first article

eediallo committed:
removes extra article

eediallo committed:
upload second article image

eediallo committed:
adds second article

eediallo committed:
adds content of second article

eediallo committed:
upload image for third article

eediallo committed:
adds third article

eediallo committed:
adds content of third article

eediallo committed:
removes articles section border

eediallo committed:
removes bar for used for skeleting the page

eediallo committed:
removes header section border

eediallo committed:
updates description of site

eediallo committed:
styles footer

eediallo committed:
styles body of webpage

eediallo committed:
makes site responsive using media

SallyMcGrath commented on my-wireframe/articles/why-do-developers-need-git.html:
```diff
@@ -0,0 +1,82 @@
+    </section>
+
+    <section>
+        <h4>5. Transparency adn Accountability</h4>
```
```suggestion
        <h4>5. Transparency and Accountability</h4>
```

I'm just making a small change to trigger deployment previews as your PR was opened before I turned this feature on.

eediallo replied:
Thanks for the update! That sounds great. Let me know if there's anything else I need to do on my end.

SallyMcGrath committed:
Update my-wireframe/articles/why-do-developers-need-git.html

SallyMcGrath commented on my-wireframe/css/index.css:
```diff
@@ -0,0 +1,81 @@
+    text-decoration: none;
+}
+
+a:hover{
```
```suggestion
a:hover, a:focus{
```

Not all users are using a mouse 🪤

eediallo replied:
Thank you! I have made the changes.

SallyMcGrath commented on my-wireframe/index.html:
```diff
@@ -0,0 +1,56 @@
+<body>
+    <header>
+        <h1>Git Essentials: A Guide to Version Control for Developers</h1>
+        <p><b>Git Essentials</b> is a beginner-friendly site  designed to explain the fundamental concepts of
```
```suggestion
        <p><cite>Git Essentials</cite> is a beginner-friendly site  designed to explain the fundamental concepts of
```

eediallo replied:
Thank you for the suggestion. I have replace the b with cite tag.

SallyMcGrath commented on my-wireframe/index.html:
```diff
@@ -0,0 +1,56 @@
+
+    <div class="articles">
+        <article>
+            <img src="images/git.webp"></img>
```
Is this a valid img tag? How would you find out?

Your website is deployed as a PR preview here https://deploy-preview-109--cyf-ufd.netlify.app/my-wireframe/

Try running Lighthouse on the page. And sending it to the [validator](https://validator.w3.org/nu/?doc=https%3A%2F%2Fdeploy-preview-109--cyf-ufd.netlify.app%2Fmy-wireframe%2F).  What advice do these tools give you?

eediallo replied:
I have added the alt tag for each image tag and reomves the closing. Now the Nu Html Checker is happy with it.

SallyMcGrath commented on my-wireframe/articles/why-do-developers-need-git.html:
```diff
@@ -0,0 +1,82 @@
+</head>
+
+<body>
+    <section>
```
Make sure you always have an h1 on your page. Here's a great page from a11y on this https://www.a11yproject.com/posts/how-to-accessible-heading-structure/

SallyMcGrath commented on my-wireframe/articles/why-do-developers-need-git.html:
```diff
@@ -0,0 +1,82 @@
+    </section>
+
+    <section>
+        <h4>1. Version Control</h4>
```
Is h4 the right heading level to begin at? What does the linked article say?

SallyMcGrath reviewed (changes requested):
Welcome to ITP! Congratulations on your first Pull Request! 🎉

It's a wonderful start, @eediallo ! You can see your work deployed on a PR preview here 

https://deploy-preview-109--cyf-ufd.netlify.app/my-wireframe/

1. You seem t It doesn't look like the layout quite matches the structure in the wireframe yet. Do you need some help with this?

2. You _can_ use a div to wrap and group several semantic elements together to help with layout. That's really the only correct use of div. If you're _describing_ content, use a semantic html element like header, section, aside, etc. If you're just grouping together to apply a css rule, then div is the right choice.

eediallo committed:
replaces b tag with cite tag in header description

eediallo committed:
adds alt attribute to each image article

eediallo committed:
removes closing img tag

eediallo committed:
adds my name as footer

eediallo committed:
focuses anchor tags for non mouse users

eediallo committed:
put footer content in a h2 tag

eediallo committed:
moves first article out of the articles section

eediallo committed:
structure first article lay out

eediallo committed:
removes first two div in first article section

eediallo committed:
updates style of article 1

eediallo committed:
removes div tag around first article

eediallo committed:
changes class name for article 2 and 3

eediallo committed:
updates article 1 style using the first-child pseudo class

eediallo committed:
updates display of article 2 and 3 in media query

eediallo committed:
adjust header section's width and first article view on small screen

eediallo committed:
replaces h3 with h1 for each article title

eediallo committed:
replaces h1 with 2 for each article title  for clear and logical hierarchy

eediallo commented:
Hi @SallyMcGrath! Thank you for the review. I updated the webpage layout to match the wireframe structure. If it's not right, I need help to understand the task better. I also made the changes you suggested.

SallyMcGrath commented:
> Hi @SallyMcGrath! Thank you for the review. I updated the webpage layout to match the wireframe structure. If it's not right, I need help to understand the task better. I also made the changes you suggested.

It looks like the wireframe to me. What do you think? Do you feel like you've completed this project? It's your opinion that matters here. I'm just here to help you get to where you want to go.

eediallo commented:
> > Hi @SallyMcGrath! Thank you for the review. I updated the webpage layout to match the wireframe structure. If it's not right, I need help to understand the task better. I also made the changes you suggested.
> 
> It looks like the wireframe to me. What do you think? Do you feel like you've completed this project? It's your opinion that matters here. I'm just here to help you get to where you want to go.

Thank you for the feedback! I appreciate your support! I'd like to ensure that I'm on the right track with this project. Could you provide more specific feedback on how well the layout aligns with the wireframe and any areas I could improve? I want to make sure I'm meeting the expectations and completing the project to its fullest potential.

eediallo committed:
adds meta description for SEO

eediallo committed:
adds explicit with and height to avoid layout shift

eediallo committed:
convert images from jpeg format to webp format to allow faster dowload and less data consumption

eediallo committed:
changes images of jpeg format to webp format

eediallo committed:
deletes jpeg format images

SallyMcGrath commented:
> > > Hi @SallyMcGrath! Thank you for the review. I updated the webpage layout to match the wireframe structure. If it's not right, I need help to understand the task better. I also made the changes you suggested.
> > 
> > 
> > It looks like the wireframe to me. What do you think? Do you feel like you've completed this project? It's your opinion that matters here. I'm just here to help you get to where you want to go.
> 
> Thank you for the feedback! I appreciate your support! I'd like to ensure that I'm on the right track with this project. Could you provide more specific feedback on how well the layout aligns with the wireframe and any areas I could improve? I want to make sure I'm meeting the expectations and completing the project to its fullest potential.

So let's take a sec to think about that. This project is of no external value. Completing it is not the point. Nobody is waiting to load this website. It's entirely a construct for _you_ to think through and explore some introductory ideas. So the thing for you to do is to set out your learning goals and then decide if you have met them.

For me, I think this project has a few goals:

1. To practice opening a PR, getting a review, and making changes based on that. The whole code review process, which you will do many times, starts here.
2. To practice reading requirements and evaluating your work against those requirements, to understand if something is "done". Software development is a process of interpreting requirements (which may come as words, a picture, a complaint, a failing test, etc), expressing those requirements as code that meets its needs, and then testing your work, both with automated tests, unit tests, and human feedback. This constant iterative process is how we develop. 
3. To describe some important first concepts in Git, which you will use many times. In particular, small scoped feature branching, which you also practice in your PR.

You might see some other goals as well. 

We could, honestly, revise just this one page for three months and still likely there would be more learning we could wring from it. We haven't really examined your CSS, or the design beyond "is it roughly the right layout". We haven't edited your writing. We haven't talked much about how this CI/CD deployment on the PR preview we're looking at works. We could go on forever! 😉

eediallo commented:
> > > > Hi @SallyMcGrath! Thank you for the review. I updated the webpage layout to match the wireframe structure. If it's not right, I need help to understand the task better. I also made the changes you suggested.
> > > 
> > > 
> > > It looks like the wireframe to me. What do you think? Do you feel like you've completed this project? It's your opinion that matters here. I'm just here to help you get to where you want to go.
> > 
> > 
> > Thank you for the feedback! I appreciate your support! I'd like to ensure that I'm on the right track with this project. Could you provide more specific feedback on how well the layout aligns with the wireframe and any areas I could improve? I want to make sure I'm meeting the expectations and completing the project to its fullest potential.
> 
> So let's take a sec to think about that. This project is of no external value. Completing it is not the point. Nobody is waiting to load this website. It's entirely a construct for _you_ to think through and explore some introductory ideas. So the thing for you to do is to set out your learning goals and then decide if you have met them.
> 
> For me, I think this project has a few goals:
> 
> 1. To practice opening a PR, getting a review, and making changes based on that. The whole code review process, which you will do many times, starts here.
> 2. To practice reading requirements and evaluating your work against those requirements, to understand if something is "done". Software development is a process of interpreting requirements (which may come as words, a picture, a complaint, a failing test, etc), expressing those requirements as code that meets its needs, and then testing your work, both with automated tests, unit tests, and human feedback. This constant iterative process is how we develop.
> 3. To describe some important first concepts in Git, which you will use many times. In particular, small scoped feature branching, which you also practice in your PR.
> 
> You might see some other goals as well.
> 
> We could, honestly, revise just this one page for three months and still likely there would be more learning we could wring from it. We haven't really examined your CSS, or the design beyond "is it roughly the right layout". We haven't edited your writing. We haven't talked much about how this CI/CD deployment on the PR preview we're looking at works. We could go on forever! 😉

Brilliant! Thank you for the feedback! I used the resources you provided, especially **_PageSpeed Insights_**, to boost the performance of the web page. It's now up from 77 to 83! Right now, I'm going to focus on other tickets, but I'll definitely come back to this project to make it even better as I learn new things and get feedback. Thanks a bunch for your time! 🙏🏻

eediallo commented:
Hello @SallyMcGrath,

This pull request still requires me to make changes as it has a "review requirement" label attached. Kindly inform me if I have overlooked any changes that you suggested.

Thank you.

moneyinthesky commented:
Hi @eediallo! If you're happy with this PR and you've addressed the comments - feel free to add a `Complete` label to it.
The "review requirements" label is there to highlight an area you could focus on in the future, and also for volunteers to understand what the common areas participants need help with.

eediallo commented:
> Hi @eediallo! If you're happy with this PR and you've addressed the comments - feel free to add a `Complete` label to it. The "review requirements" label is there to highlight an area you could focus on in the future, and also for volunteers to understand what the common areas participants need help with.

Hi @moneyinthesky thank you for clarifying this.
